plugin/plugins/filesystem/nodes/ipfs.go

```diff
@@ -0,0 +1,222 @@
+package fsnodes
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/djdv/p9/p9"
+	files "github.com/ipfs/go-ipfs-files"
+	logging "github.com/ipfs/go-log"
+	coreiface "github.com/ipfs/interface-go-ipfs-core"
+	corepath "github.com/ipfs/interface-go-ipfs-core/path"
+)
+
+// IPFS exposes the IPFS API over a p9.File interface
+// Walk does not expect a namespace, only its path argument
+// e.g. `ipfs.Walk([]string("Qm...", "subdir")` not `ipfs.Walk([]string("ipfs", "Qm...", "subdir")`
+type IPFS struct {
+	IPFSBase
+	ipfsHandle
```

Comment:
We also need to close the file/directory inside `ipfsHandle`.

Reply:
If one is assigned to the reference, we should now close it.
https://github.com/ipfs/go-ipfs/commit/21b2d2cf1b13db5076405173d02a803f4a8c241f#diff-6c2c294fc60db12a68ce40457ff3ea5eR67